airbyte-integrations/connectors/source-sumologic/source_sumologic/source.py

```diff
@@ -0,0 +1,106 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+import datetime
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+
+from .client import Client
+
+
+# Basic full refresh stream
+class SumologicStream(Stream, ABC):
+
+    primary_key = "_messageid"
+
+    def __init__(self, client: Client, config: Mapping[str, Any]) -> None:
+        super().__init__()
+        self.client = client
+        self.config = config
+
+
+# Basic incremental stream
+class IncrementalSumologicStream(SumologicStream, ABC):
+
+    # Not confident with the order returned by sumologic, so don't check in until all data are read
+    state_checkpoint_interval = None
```

Comment:
This not recommended to not checkpoint your stream in an incremental mode, otherwise use Full refresh streams... I suggest that you implement `stream_slices` to handle a smart checkpointing if you're not confident with the returned order. Doc [here](https://docs.airbyte.io/connector-development/cdk-python/stream-slices).